components/src/dynamo/common/multimodal/image_loader.py

```
@@ -213,32 +217,54 @@ async def load_image_batch(
+                logger.debug(
+                    f"UUID-only image slot (cache lookup): {item[UUID_ONLY_VARIANT_KEY]}"
+                )
 
-        # Process images in parallel
+        # Process futures in parallel; weave None back in for uuid-only slots
         results = await asyncio.gather(*image_futures, return_exceptions=True)
-        loaded_images = []
+        loaded_images: list[Any] = []
         collective_exceptions = ""
-        for media_item, result in zip(image_mm_items, results):
+        for media_item, fut_idx in zip(image_mm_items, slot_to_future_idx):
```

Ruff (0.15.12)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)